src/components/PopupWithForm.js

```diff
@@ -32,9 +32,26 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
+      //this.close();
     });
     //The setEventListeners() method of the PopupWithForm class should add a submit event listener to the form
     //and call the setEventListeners() method of the parent class
   }
+
+  renderLoadingSave(loading) {
+    const submitButton = this._popupForm.querySelector(".modal__button");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.